### PR TITLE
Add course deprecation page

### PR DESCRIPTION
**Pull request by bencodeorg**
Adds a new page to redirect to for deprecated courses. Currently not linked to anywhere, follow-up work will do the actual redirection.

Mostly modeled off of our existing not_found page -- tried to make a couple of improvements to what's on that page:
- used `<a>` tag with "button" styling, rather than a `button` tag nested in an `a` tag, which is what we have on not_found. I believe this is the semantic HTML for a link that looks like a button, but LMK if anyone knows better!
- put this styling in a place (application.scss) that could be reused in similar contexts.

image

## Links

- [jira ticket](https://codedotorg.atlassian.net/browse/SL-894)

## Testing story

I tested manually (screenshot above) by setting our 404 page to show this view temporarily.

### CODE REVIEW

File: dashboard/app/assets/stylesheets/application.scss [hunk 1]
```diff
@@ -3967,4 +3967,39 @@ html[dir=rtl] {
   }
 }
 
+a {
```
Comment (bencodeorg) [inline dashboard/app/assets/stylesheets/application.scss:3970]: @levadadenys I haven't done much CSS in the context of our new design system, so LMK what you think of this -- my thought was to add a new class for a tags that look like buttons with the updated design system. Might be a preferred way to do this, let me know if so!

**Review by sanchitmalhotra126 — APPROVED** [review]
Code LGTM! I'll defer to @levadadenys for branding feedback

File: dashboard/app/assets/stylesheets/errors.scss [hunk 2]
```diff
@@ -86,3 +86,17 @@
...
+.deprecated-course-page {
+  margin-bottom: 100px;
+
+  .deprecated-course-child {
+    display: inline-block;
+    max-width: 500px;
+  }
+
+  h1 {
+    color: $neutral_dark;
```
Comment (levadadenys) [inline dashboard/app/assets/stylesheets/errors.scss:99]: I would also add   `font-family: $barlowSemiCondensed-semibold;` here, so it'll use the same fontFamily that our designSystem uses

**Review by levadadenys — APPROVED** [review]
LGTM in general!
Have just one not urgent suggestion.

--

Also, makes me think we might wanna implement design system for haml files or use react instead of haml files(?) (Currently we do have design system for react code in `/apps` and haml code in `/pegasus`, not in `/apps`)

File: dashboard/app/assets/stylesheets/errors.scss (hunk 2 above)
Comment (levadadenys) [reply dashboard/app/assets/stylesheets/errors.scss:99]: image

Comment (breville) [reply dashboard/app/assets/stylesheets/errors.scss:99]: Isn't that font just used for headings, rather than multi-sentence paragraphs?

Comment (levadadenys) [reply dashboard/app/assets/stylesheets/errors.scss:99]: Yes it is. But it's `h1`, not `p` tag currently

Comment (levadadenys) [reply dashboard/app/assets/stylesheets/errors.scss:99]: If it's supposed to look like `p`, not `h1` then I'm ok with it

Comment (breville) [reply dashboard/app/assets/stylesheets/errors.scss:99]: Yes, I think the styling issue is that we want a big font, but not necessarily a heading.  Do the design guidelines have a recommendation for that?

File: dashboard/app/views/errors/deprecated_course.html.haml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+.not-found-page.deprecated-course-page
+  .error-parent
+    .deprecated-course-child
+      %img{src: '/shared/images/happy_penguin.png', alt: 'A happy cartoon penguin'}
+      %h1= I18n.t('course_deprecated.title')
```
Comment (mgc1194) [inline dashboard/app/views/errors/deprecated_course.html.haml]: Awesome! i18n from the design! If you have not done it, I'd suggest checking the page renders as expected with right-to-left languages

File: dashboard/app/assets/stylesheets/errors.scss (hunk 2 above)
Comment (levadadenys) [reply dashboard/app/assets/stylesheets/errors.scss:99]: We agreed that we can have any html semantic tag look like any 'visualAppearance'.
Meaning that `h1` can look like `bodyOneText` and vice versa.
We can stick with this version, or you can copy paste the styles from `visualAppearance` in `typography.module.scss` you want to use for this text

File: dashboard/app/views/errors/deprecated_course.html.haml (hunk 3 above)
Comment (bencodeorg) [reply dashboard/app/views/errors/deprecated_course.html.haml]: I have not, thanks for the suggestion! Will do.

File: dashboard/app/assets/stylesheets/errors.scss (hunk 2 above)
Comment (bencodeorg) [reply dashboard/app/assets/stylesheets/errors.scss:99]: I switched this to a `p` tag to be more semantically appropriate, with "big font" styling.

File: dashboard/app/views/errors/deprecated_course.html.haml (hunk 3 above)
Comment (bencodeorg) [reply dashboard/app/views/errors/deprecated_course.html.haml]: Chatted in Slack, Mario confirmed RTL looked good!

**Pull request merged**